src/webjure/json_schema/validator.clj

```diff
@@ -109,35 +128,34 @@
   (when enum
     (let [allowed-values (into #{} enum)]
       (when-not (allowed-values data)
-        {:error :invalid-enum-value
-         :data data
+        {:error          :invalid-enum-value
+         :data           data
          :allowed-values allowed-values}))))
-  
+
 (defmethod validate-by-type "string"
-  [_ data _]
-  (when-not (string? data)
-    {:error :wrong-type :expected :string :data data}))
+  [schema data _]
+  (if (not (string? data))
+    {:error :wrong-type :expected :string :data data}
+    (let [format (get schema "format")]
+      (when (and format (= format "date-time") (nil? (time/from-string data)))
+        {:error :wrong-type :expected :date :data data}))))
 
 (defmethod validate-by-type "array"
-  [{item-schema "items"} data options]
+  [schema data options]
   (if-not (sequential? data)
     {:error :wrong-type :expected :array-like :data data}
-    (loop [errors []
-           i 0
-           [item & items] data]
-      (if-not item
-        (if (empty? errors)
-          nil
-          {:error :array-items
-           :data data
-           :items errors})
-        (let [item-error (validate-by-type item-schema item options)]
-          (recur (if item-error
-                   (conj errors (assoc item-error
-                                  :position i))
-                   errors)
-                 (inc i)
-                 items))))))
+    (do
+      (let [min-items (get schema "minItems")
+            max-items (get schema "maxItems")
+            item-schema (get schema "items")
+            uniqueItems (get schema "uniqueItems")]
+        (if (and min-items (> min-items (count data)))
```

Comment:
Use cond instead of multiple nested ifs?

the do that wraps the let is unnecessary